### PR TITLE
Add keep-uid argument to preserve the uid provided in file

### PR DESCRIPTION
**Pull request by vrymar**
Add keep-uid argument to preserve the uid which is defined in the import dashboard file.
Provides solution for issue:
- GH-23

### CODE REVIEW

**Comment by amotl** [comment]
Hi. Thanks for your contribution. After installing the optional `develop` dependency packages using `pip install --editable='.[develop]'`, you can format and lint the code base using `poe format` and `poe lint`.

`poe check` is a catch-all command to invoke both the code formatter checks, the linter, and software tests, like on CI. If that signals success, CI should do as well, as long as your patch doesn't add any other side effects that are only satisfied on the local sandbox environment.

File: grafana_import/grafana.py [hunk 1]
```diff
@@ -305,18 +308,23 @@ def import_dashboard(self, dashboard: t.Dict[str, t.Any]) -> bool:
...
             # ** case d) send a copy to existing dash : update existing
             elif new_dash["folderId"] == old_dash["folderId"]:
-                if "uid" not in new_dash["dashboard"] or new_dash["dashboard"]["uid"] != old_dash["uid"]:
+                if ("uid" not in new_dash["dashboard"]
+                        or new_dash["dashboard"]["uid"] != old_dash["uid"]
+                        or new_dash["dashboard"]["id"] != old_dash["id"]):
                     if self.overwrite:
-                        new_dash["dashboard"]["uid"] = old_dash["uid"]
+                        if not self.keep_uid:
+                            new_dash["dashboard"]["uid"] = old_dash["uid"]
```
Comment (amotl) [inline grafana_import/grafana.py:316]: Is this fragment correct, or should the check be inverted?
```suggestion
                        if self.keep_uid:
                            new_dash["dashboard"]["uid"] = old_dash["uid"]
```

Comment (vrymar) [reply grafana_import/grafana.py:316]: @amotl , `-keep_uid` argument is used to preserve uid provided in dashboard JSON file under any conditions. The line `new_dash["dashboard"]["uid"] = old_dash["uid"]` overwrites JSON uid by the existing dashboard uid  So, to keep the JSON uid, when we provide `-keep-uid` arg even with `-overwrite` arg, we need to set:
`if not self.keep_uid:
        new_dash["dashboard"]["uid"] = old_dash["uid"]`
 Hope I expressed that clearly :). If not, you are welcome to ask more questions.

**Comment by vrymar** [comment]
> Hi. Thanks a stack for your contribution. After installing the optional `develop` dependency packages using `pip install --editable='.[develop]'`, you can format and lint the code base using `poe format` and `poe lint`.
> 
> `poe check` is a catch-all command to invoke both the code formatter checks, the linter, and software tests, like on CI. If that signals success, CI should do as well, as long as your patch doesn't add any other side effects that are only satisfied on the local sandbox environment.
> 
> In this spirit, may I ask you to adjust your patch, and, after rebasing on top of `main`, also add a corresponding changelog item to `HISTORY.md`? Thanks in advance!

@amotl, thank you for the detailed explanation. `poe format`, `poe lint` passed, but `poe check` failed during tests execution. Also I've noticed, that it fails on `main` branch as well.

```
ERROR tests/test_cli.py::test_import_dashboard_success[config-yes] - AssertionError: Not all requests have been executed [('GET', 'http://localhost:3000/api/health'), ('GET', 'http://localhost:3000/api/search?type=dash-db&limit=5000'), ('POST', 'http://localhost:3000/api/dashboards/db')]
ERROR tests/test_cli.py::test_import_dashboard_success[config-no] - AssertionError: Not all requests have been executed [('GET', 'http://localhost:3000/api/health'), ('GET', 'http://localhost:3000/api/search?type=dash-db&limit=5000'), ('POST', 'http://localhost:3000/api/dashboards/db')]
ERROR tests/test_core.py::test_import_dashboard_success[config-yes] - AssertionError: Not all requests have been executed [('GET', 'http://localhost:3000/api/health'), ('GET', 'http://localhost:3000/api/search?type=dash-db&limit=5000'), ('POST', 'http://localhost:3000/api/dashboards/db')]
ERROR tests/test_core.py::test_import_dashboard_success[config-no] - AssertionError: Not all requests have been executed [('GET', 'http://localhost:3000/api/health'), ('GET', 'http://localhost:3000/api/search?type=dash-db&limit=5000'), ('POST', 'http://localhost:3000/api/dashboards/db')]
FAILED tests/test_cli.py::test_export_dashboard_success[config-yes] - assert None
 +  where None = <function match at 0x00000279B2C514E0>("OK: Dashboard 'foobar' exported to: ./foobar_\\d+.json", "OK: Dashboard 'foobar' exported to: .\\foobar_20241014132027.json\n")
 +    where <function match at 0x00000279B2C514E0> = re.match
```

**Comment by amotl** [comment]
> but `poe check` failed during tests execution. Also I've noticed, that it fails on main branch as well.

Oops. Thanks for also fixing that. As CI signals success, I am merging your patch now.

**Comment by vitaliirymar** [comment]
Actually, I haven't fixed the failed tests (

**Pull request merged**